src/core/style/mixins.scss

```diff
@@ -337,3 +337,56 @@
     @include rtl($prop, $ltr-dense, $rtl-dense);
   }
 }
+
+// Only use when in row layout
+@mixin when-layout-row($element) {
+  @media (max-width: $layout-breakpoint-xs - 1) {
+    .layout-row:not(.layout-xs-column),
+    .layout-xs-row {
+      & > #{$element} { @content; }
+    }
+  }
+  @media (min-width: $layout-breakpoint-xs) and (max-width: $layout-breakpoint-sm - 1) {
+    .layout-row:not(.layout-gt-xs-column),
+    .layout-gt-xs-row,
+    .layout-sm-row {
+      &:not(.layout-sm-column) > #{$element} { @content; }
+    }
+  }
+  @media (min-width: $layout-breakpoint-sm) and (max-width: $layout-breakpoint-md - 1) {
+    .layout-row:not(.layout-gt-xs-column):not(.layout-gt-sm-column),
+    .layout-gt-xs-row:not(.layout-gt-sm-column),
+    .layout-gt-sm-row,
+    .layout-md-row {
+      &:not(.layout-md-column) > #{$element} { @content; }
+    }
+  }
+  @media (min-width: $layout-breakpoint-md) and (max-width: $layout-breakpoint-lg - 1) {
+    .layout-row:not(.layout-gt-xs-column):not(.layout-gt-sm-column):not(.layout-gt-md-column),
+    .layout-gt-xs-row:not(.layout-gt-sm-column):not(.layout-gt-md-column),
+    .layout-gt-sm-row:not(.layout-gt-md-column),
+    .layout-gt-md-row,
+    .layout-lg-row {
+      &:not(.layout-lg-column) > #{$element} { @content; }
+    }
+  }
+  @media (min-width: $layout-breakpoint-lg) {
+    .layout-row:not(.layout-gt-xs-column):not(.layout-gt-sm-column):not(.layout-gt-md-column),
+    .layout-gt-xs-row:not(.layout-gt-sm-column):not(.layout-gt-md-column),
+    .layout-gt-sm-row:not(.layout-gt-md-column),
+    .layout-gt-md-row,
+    .layout-gt-lg-row,
+    .layout-xl-row {
+      &:not(.layout-gt-lg-column):not(.layout-xl-column) > #{$element} { @content; }
+    }
+  }
+}
+
+// Auto insert object margin
+@mixin auto-horizontal-margin($selector) {
+  @include when-layout-row($selector) {
+    &:not(last-child) {
```

Comment:
I believe that this should be `:last-child` to actually work properly.

That said, in my testing, that results in this:
![Screen Shot 2020-07-27 at 15 00 05](https://user-images.githubusercontent.com/3506071/88580704-03d15900-d01a-11ea-9883-6d97e0eea800.png)

Using `:first-child` seems to give us what we want:
![Screen Shot 2020-07-27 at 15 00 34](https://user-images.githubusercontent.com/3506071/88580732-0fbd1b00-d01a-11ea-8c52-68782248c7f7.png)

Reply:
I think it should be `:last-child` as you noted, or else it would target against `<last-child>` tag name, which would be wrong. I would check against a row with 3 items to ensure the auto margins work. It might look right with `:first-child` because they're only 2 items.

Reply:
The rule, in theory, should pad `$default-horizontal-margin` at the start (left on LTR). Left padding should appear on every item, except the first one, so `:not(:first-child)` makes sense. 

Maybe I did a last minute change and had it apply margin at the end originally, and failed to adjust. So yeah, `:not(:first-child)` is good.